smartclide-monitoring/src/main/java/eu/smartclide/contexthandling/services/GitRestCallService.java

```diff
@@ -0,0 +1,183 @@
+package eu.smartclide.contexthandling.services;
+
+/*-
+ * #%L
+ * SmartCLIDE Monitoring
+ * %%
+ * Copyright (C) 2015 - 2022 ATB – Institut für angewandte Systemtechnik Bremen GmbH
+ * %%
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ * #L%
+ */
+
+import com.google.gson.*;
+import de.atb.context.monitoring.models.GitMessage;
+import de.atb.context.monitoring.models.GitMessageHeader;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.InvalidPathException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+
+import com.google.gson.JsonArray;
+
+public class GitRestCallService {
+
+    private static final Logger logger = LoggerFactory.getLogger(GitRestCallService.class);
+
+    private final String accessTokenParam;
+    private final String baseUri = "https://gitlab.atb-bremen.de/api/v4/projects/";
```

Comment:
`baseUri`should also be a constructor parameter because the GitLab instance to be used will be configured through the monitoring configuration.

The part `/api/v4/projects` can remain a private static final class variable.